src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_IIntellisenseBuildTarget.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Logging;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal partial class AbstractProject : IIntellisenseBuildTarget
+    {
+        private static readonly object s_diagnosticKey = new object();
+        private static readonly FieldInfo s_fieldInfo = typeof(BuildManager).GetField("_buildParameters", BindingFlags.GetField | BindingFlags.Instance | BindingFlags.NonPublic);
+
+        // REVIEW: can temp path changed while process running?
+        private static readonly string GeneralLogFile = $"{Path.GetTempPath()}\\<guid>.designtime.log";
+
+        // set default to true so that we maintain old behavior when project system doesn't
+        // implement IIntellisenseBuildTarget
+        private bool _isIntellisenseBuildSucceeded = true;
+        private string _intellisenseBuildFailureReason = null;
+
+        public void SetIntellisenseBuildResult(bool succeeded, string reason)
+        {
+            // set intellisense related info
```

Comment:
all VS interface is UI only. looks like none of other VS interface implementations in the class assert on UI thread. probably implicitly assumed?